pkg/queue/manager.go

```diff
@@ -88,9 +151,19 @@ func (m *Manager) UpdateQueue(q *kueue.Queue) error {
 	defer m.Unlock()
 	qImpl, ok := m.queues[Key(q)]
 	if !ok {
-		return errQueueDoesntExist
+		return queueDoesNotExistErr
+	}
```

Comment:
Shall we allow user to change the ClusterQueue of the Queue？
I think it is better to prohibit this operation？

Reply:
So the alternative would be to delete the existing Q and create a new one with the same name. Depending on how fast this happens and how busy the apiserver and kueue are, this set of operations could be collapsed into a single update. So we need to handle it.

Reply:
I'm not sure if it still collapses the delete and create into a single update event even when the object's UID is different. 

If it will, would it be better to use UID as a key to avoid this this? The reason I have this question is because I'm worried about what workloads are already scheduled and how you're going to handle them if the queue switches to another capacity?

Reply:
I know that the raw go-client does collapse the events even if the UIDs are different. I don't know about the controller-runtime's client. Sure, we can deal with this in the event handler instead, but I'm not too concerned.

>  The reason I have this question is because I'm worried about the workloads which are already scheduled and how you're going to handle them if the queue switches to another capacity?

I think we shouldn't disturb the running workloads, just the pending ones.

Reply:
> I think we shouldn't disturb the running workloads, just the pending ones.

got it.